tests/test_workers/test_pkg_managers/test_npm.py

```diff
@@ -0,0 +1,214 @@
+                "version": "0.803.26",
+            }
+        ],
+        "@angular/animations": [
+            {
+                "bundled": False,
+                "dev": False,
+                "name": "@angular/animations",
+                "type": "npm",
+                "version": "8.2.14",
+            }
+        ],
+        "rxjs": [
+            {"bundled": False, "dev": True, "name": "rxjs", "type": "npm", "version": "6.4.0"},
+            {"bundled": False, "dev": False, "name": "rxjs", "type": "npm", "version": "6.5.5"},
+        ],
+        "tslib": [
+            {"bundled": False, "dev": False, "name": "tslib", "type": "npm", "version": "1.11.1"}
+        ],
+    }
+
+
+def test_get_deps_non_registry_dep():
+    package_lock_deps = {
+        "@angular/animations": {
+            "version": "8.2.14",
+            "resolved": "https://registry.npmjs.org/@angular/animations/-/animations-8.2.14.tgz",
+            "integrity": (
+                "sha512-3Vc9TnNpKdtvKIXcWDFINSsnwgEMiDmLzjceWg1iYKwpeZGQahUXPoesLwQazBMmxJzQiA"
+                "4HOMj0TTXKZ+Jzkg=="
+            ),
+            "requires": {"tslib": "^1.9.0"},
+        },
+        "tslib": {
+            "version": "github:microsoft/tslib#c1f87f79190d61e1e4ca24af03894771cdf1aef9",
+            "from": "github:microsoft/tslib",
+        },
+    }
+    expected = (
+        "The lock file contains a dependency not from the npm registry. This is not yet supported."
+    )
+    with pytest.raises(CachitoError, match=expected):
+        npm._get_deps(package_lock_deps, {})
+
+
+def test_get_package_and_deps(package_lock_deps, package_and_deps):
+    package_lock = {"name": "han_solo", "version": "5.0.0", "dependencies": package_lock_deps}
+    mock_open = mock.mock_open(read_data=json.dumps(package_lock))
+    with mock.patch("cachito.workers.pkg_managers.npm.open", mock_open):
+        package, deps = npm.get_package_and_deps(
+            "/tmp/cachito-bundles/1/temp/app/package-lock.json"
+        )
+
+    deps = sorted(deps, key=operator.itemgetter("name", "version"))
+
+    expected_package, expected_deps = package_and_deps
+    assert package == expected_package
+    assert deps == expected_deps
+
+
+@pytest.mark.parametrize("shrink_wrap, package_lock", ((True, False), (True, True), (False, True)))
+@mock.patch("cachito.workers.pkg_managers.npm.os.path.exists")
+@mock.patch("cachito.workers.pkg_managers.npm.get_package_and_deps")
+@mock.patch("cachito.workers.pkg_managers.npm.download_dependencies")
+def test_resolve_npm(mock_dd, mock_gpad, mock_exists, shrink_wrap, package_lock, package_and_deps):
+    mock_exists.side_effect = [shrink_wrap, package_lock]
+    mock_gpad.return_value = package_and_deps
+
+    expected_package, expected_deps = package_and_deps
+
+    src_path = "/tmp/cachito-bundles/temp/1/app"
+    package, deps = npm.resolve_npm(src_path, {"id": 1})
+
+    # Note that the deps returned by the get_package_and_deps function were modified as part of the
```

Comment:
Do we need to make a deep copy of  `expected_package` and `expected_deps` to ensure we're comparing the right thing?